src/main/java/seedu/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,45 @@
+package seedu.duke.tasks;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline class
+ */
+public class Deadline extends Task {
+    private LocalDateTime byDate;
+
+    /**
+     * Deadline constructor
+     *
+     * @param description user input
+     * @param byDate dateline of task
+     * @param isMarked is task marked
+     */
+    public Deadline(String description, LocalDateTime byDate, boolean isMarked) {
+        super(description, isMarked);
+        this.byDate = byDate;
+    }
+
+    /**
+     * Formats description for writing to duke.txt
+     *
+     * @return proper string format
+     */
+    public String writeFormat() {
+        int isDone = super.isMarked() ? 1 : 0;
```

Comment:
Currently, the name for this variable sounds like a boolean. Perhaps we can change the variable type to boolean or we can rename the variable to something more affirmative like "done".

Reply:
got it, changed naming to done